src/components/ExperimentalFeaturesMenu.tsx

```diff
@@ -41,20 +50,32 @@ export function ExperimentalFeaturesMenu() {
                     <button
                       className="flex items-center gap-2 m-0 py-1.5 px-2 cursor-pointer hover:bg-chalkboard-20 dark:hover:bg-chalkboard-80 border-none text-left"
                       onClick={() => {
-                        setExperimentalFeatures(level)
-                          .then((result) => {
-                            if (err(result)) {
-                              toast.error(
-                                `Failed to set file experimental features level: ${result.message}`
-                              )
-                              return
-                            }
-
-                            toast.success(
-                              `Updated file experimental features level to ${level.type}`
-                            )
+                        const newAst = setExperimentalFeatures(level)
```

Comment:
Changing this to act like other code mods, returning the new ast to be updated the regular way, instead of just a code execution.